File: output/CodeGen/codenet/C++/Python/s501065052.py
def main():
    N = int(input())
    mem = [0] * (N + 1)
    for i in range(1, N + 1):
        P = int(input())
        mem[P] = i
    maxcnt = 1
    cnt = 1
    for i in range(1, N):
        if mem[i] < mem[i + 1]:
            cnt += 1
        else:
            maxcnt = max(maxcnt, cnt)
            cnt = 1
    print(N - maxcnt)

if __name__ == '__main__':
    main()

File: output/CodeGen/codenet/C++/Java/s501065052.java
public class s501065052 {
    public int findLengthOfLCIS(int[] nums) {
        int len = nums.length;
        if (len == 0) return 0;
        int[] mem = new int[len];
        int max = 1;
        int cnt = 1;
        for (int i = 1; i < len; i++) {
            if (nums[i] < nums[i - 1]) {
                cnt++;
            } else {
                max = Math.max(max, cnt);
                cnt = 1;
            }
            mem[i] = cnt;
        }
        max = Math.max(max, cnt);
        cnt = 1;
        for (int i = len - 1; i >= 1; i--) {
            if (nums[i] > nums[i - 1]) {
                cnt++;
            } else {
                max = Math.max(max, cnt);
                cnt = 1;
            }
            mem[i] = cnt;
        }
        return len - max;
    }
}